### ISSUE TITLE
Support Paren Mode 

### ISSUE THREAD

**Issue by bhurlow** [bhurlow/vim-parinfer#34]

**Comment by osfameron** [comment]
Ah, I was just about to submit a question about the demo 'Preserve indentation when editing in "Paren Mode"' from https://shaunlebron.github.io/parinfer/ not working, but that is reliant on Paren Mode presumably?

(Thanks for contributing this module btw, this looks like it may be much nicer than vim-sexp for me.  I actually like that parens are only calculated on ESC, as it lets you type what *you* wanted, and only attempts to infer after you're done.)

**Comment by bhurlow** [comment]
Yep, the vim and python implementations should support, just haven't wired up Paren mode yet, shouldn't be too tricky

**Comment by midnio** [comment]
Actually this is just really needed for this project.

**Comment by bhurlow** [comment]
@midnio can you provide more info on your use case? I believe that "paren" mode may be deprecated in favor of "smart" mode coming in future versions

**Comment by s-ol** [comment]
@bhurlow seems that "Smart Mode" is a failed-ish experiment: https://github.com/shaunlebron/parinfer/tree/master/lib#status-update-2019-smart-mode

Would be nice to see Paren mode in vim-parinfer for sure!
